### ISSUE TITLE
Whitespace skipping breaks matches unrelated to whitespaces

### ISSUE THREAD

**Issue by Hi-Angel** [pyparsing/pyparsing#253]
Given the default behavior of whitespace skipping, it means that `LineStart() + Regexp('word')` should match both of: 1. `'word'` and 2. `'   word'` *(note the leading spaces)* strings. However 2 doesn't match. One might think, perhaps it is a corner case where whitespace is singificant — but no, `LineStart() + Regexp('\s+')` will not match 2 either.

## Steps to reproduce *(in terms of terminal commands)*

```python
 λ cat test2.py
from pyparsing import LineStart, Regex

r = LineStart() + Regex('(\s+|word)')
s = '  word'
print([r for r in r.scanString(s)])
 λ python test2.py
[]
```

### Expected

The list of matches is not empty

### Actual

The list of matches is empty

### Known workarounds

Disabling whitespace skipping as in the following code makes it work.

```python
from pyparsing import LineStart, Regex, ParserElement

ParserElement.setDefaultWhitespaceChars('')
r = LineStart() + Regex('(\s+|word)')
s = '  word'
print([r for r in r.scanString(s)])
```

## Version

`2.4.7`

**Comment by Hi-Angel** [comment]
Oh, while trying to workaround it I figured, the problem is more sever, it's not even related to `Regexp` class. The following code fails to find a match too:

```python
from pyparsing import LineStart

r = LineStart() + 'word'
s = '  word'
print([r for r in r.scanString(s)])
```

**Comment by Hi-Angel** [comment]
In case someone asks: `LineStart` per see seems not to be affected by whitespace skipping, because the testcases do work if you remove leading whitespace from the string. That is, `LineStart` does match despite that whitespace skipping is enabled.

**Comment by ptmcg** [comment]
Released in 3.0.0rc2
